typing/ctype.ml

```diff
@@ -4470,10 +4462,9 @@ let match_class_types ?(trace=true) env pat_sch subj_sch =
       TypePairs.add type_pairs (self1, self2);
       (* Always succeeds *)
       moregen true type_pairs env row1 row2;
```

Comment:
There is a potential hidden read to `current_env` there:  `moregen` ⇒ `enter_poly_for` ⇒ `enter_poly` ⇒  `newtype`

Reply:
Thank you for spotting these.  Those [newty]s are changed to [newgenty].

Reply:
Ah yes, I missed the fact the level of those artificial type nodes don't matter.